clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -384,12 +384,10 @@ public void commitAsync(Map<TopicPartition, OffsetAndMetadata> offsets, OffsetCo
         final OffsetCommitCallback commitCallback = callback == null ? new DefaultOffsetCommitCallback() : callback;
         future.whenComplete((r, t) -> {
             if (t != null) {
-                commitCallback.onComplete(offsets, new KafkaException(t));
+                commitCallback.onComplete(offsets, (Exception) t);
             } else {
                 commitCallback.onComplete(offsets, null);
             }
-        }).exceptionally(e -> {
-            throw new KafkaException(e);
```

Comment:
I had a question about this code (even from before this change)—On which thread will the user's callback be executed?

I'm pretty sure that the event's `Future` is completed on the background thread, right? If so, that means that the callback passed to `whenComplete` is going to be completed on the background thread. And since it then calls the user's callback, it will also be called on the background thread.

Thoughts?

Reply:
thanks for catching this. I think you are right.